playlist_souffle/use_cases/souffle_playlist.py

```diff
@@ -0,0 +1,36 @@
+"""Module for the souffle_playlist use case."""
+
+from datetime import datetime as dt
+from playlist_souffle.definitions.playlist import Playlist
+import playlist_souffle.entities.souffler as souffler_entity
+import playlist_souffle.entities.track as track_entity
+
+def souffle_playlist(spotify, playlist_uri, user_id, shuffle_by):
```

Comment:
in pseudocode this function is just:
```
* fetch playlist from spotify service
* extract tracks and collection ids
* fetch collection tracks from spotify service
* generate souffled playlist
* save new playlist to spotify service

Reply:
Are you saying this should be in the docstring?

Reply:
I prefer the list / dict comprehension but i had to make it a batch call so i could use the concurrent futures in fetching (/later change to a batch api request). The optimization was so effective (reduced from 9s to 3s) it seems worth it

Reply:
I think fetch_related_tracks_by_track clarifies it enough